core/src/main/java/org/kiwix/kiwixmobile/core/extensions/ViewExtensions.kt

```diff
@@ -25,6 +25,20 @@ import androidx.annotation.ColorInt
 import com.google.android.material.snackbar.BaseTransientBottomBar
 import com.google.android.material.snackbar.Snackbar
 
+@SuppressLint("ShowToast")
+fun View.snack(
```

Comment:
@Sagar0-0 I understand what you want to do, for that, we should not change this function, we should write code that can be reuseable. we can define the `actionTextColor` as null, and remove the `actionTextColor` parameter from the `CoreReaderFragment` and then it will do the same what you want to achieve. Here is the code snippet you can follow.


```kotlin
@SuppressLint("ShowToast")
fun View.snack(
  stringId: Int,
  anchor: View? = null,
  actionStringId: Int? = null,
  actionClick: (() -> Unit)? = null,
  @ColorInt actionTextColor: Int? = null
) {
  Snackbar.make(
    this, stringId, Snackbar.LENGTH_LONG
  ).apply {
    actionStringId?.let { setAction(it) { actionClick?.invoke() } }
    actionTextColor?.let(::setActionTextColor)
    anchor?.let {
      anchorView = anchor
      addCallback(object : BaseTransientBottomBar.BaseCallback<Snackbar>() {
        override fun onDismissed(transientBottomBar: Snackbar?, event: Int) {
          transientBottomBar?.anchorView = null
        }
      })
    }
  }.show()
}

@SuppressLint("ShowToast")
fun View.snack(
  message: String,
  anchor: View,
  actionStringId: Int? = null,
  actionClick: (() -> Unit)? = null,
  @ColorInt actionTextColor: Int? = null
) {
  Snackbar.make(
    this, message, Snackbar.LENGTH_LONG
  ).apply {
    actionStringId?.let { setAction(it) { actionClick?.invoke() } }
    actionTextColor?.let(::setActionTextColor)
    anchorView = anchor
    addCallback(object : BaseTransientBottomBar.BaseCallback<Snackbar>() {
      override fun onDismissed(transientBottomBar: Snackbar?, event: Int) {
        transientBottomBar?.anchorView = null
      }
    })
  }.show()
}

fun View.snack(stringId: Int) {
  snack(stringId, null, null)
}

```

Reply:
I agree,  this way is more accurate. But, what is the requirement of the last function with just one param, I saw it is no different from the first one?

Reply:
The same goes for the second snack function with anchor parameter, I think having only one snack function (the first one) covers all the use cases we are having.

Reply:
@Sagar0-0 Thanks for pointing out this, it was introduced a long time ago and definitely that time it has some purpose, but now it is redundant we can remove it.

Reply:
> The same goes for the second snack function with anchor parameter, I think having only one snack function (the first one) covers all the use cases we are having.

@Sagar0-0 The first two snack bars have a difference. See the first parameter of both snack bars. The first one takes the `stringId`, and the second one takes the `string`. We did this to maintain the separate concern.

In my last comment I was talking about the below function it is now redundant.

```kotlin
fun View.snack(stringId: Int) {
  snack(stringId, null, null)
}
```

Reply:
> The same goes for the second snack function with anchor parameter, I think having only one snack function (the first one) covers all the use cases we are having.

@Sagar0-0 The first two snack bars have a difference. See the first parameter of both snack bars. The first one takes the stringId, and the second one takes the string. We did this to maintain the separate concern.

In my last comment I was talking about the below function it is now redundant.

```kotlin
fun View.snack(stringId: Int) {
  snack(stringId, null, null)
}
```